src/main/haskell/kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -61,20 +67,58 @@ symbolVerifiers =
       , Builtin.verifySymbol Bool.assertSort [trivialVerifier, trivialVerifier])
     , ("KEQUAL.neq"
       , Builtin.verifySymbol Bool.assertSort [trivialVerifier, trivialVerifier])
+    , ("KEQUAL.ite", iteVerifier)
     ]
   where
     trivialVerifier :: Builtin.SortVerifier
     trivialVerifier = const $ const $ Right ()
 
-{- | @builtinFunctions@ defines the hooks for @KEQUAL.eq@ and @KEQUAL.neq@
-which can take arbitrary terms (of the same sort) and check whether they are
-equal or not, producing a builtin boolean value.
+    iteVerifier :: Builtin.SymbolVerifier
+    iteVerifier
+        findSort
+        SentenceSymbol
+            { sentenceSymbolSorts = sorts
+            , sentenceSymbolResultSort = result
+            }
+      =
+        Kore.Error.withContext "In argument sorts" $
+            case sorts of
+                [firstSort, secondSort, thirdSort] -> do
+                    Kore.Error.koreFailWhen
+                        (arity /= 3)
+                        ("Expected 3 arguments, found " ++ show arity)
```

Comment:
You don't need this arity check anymore, right? If `arity /= 3` we would be in the other arm of the `case`.